api/v1alpha08/sonataflowplatform_types.go

```diff
@@ -45,7 +45,23 @@ type SonataFlowPlatformSpec struct {
 	// Only workflows without the `sonataflow.org/profile: dev` annotation will be configured to use these service(s).
 	// Setting this will override the use of any cluster-scoped services that might be defined via `SonataFlowClusterPlatform`.
 	// +optional
-	Services *ServicesPlatformSpec `json:"services,omitempty"`
+	Services ServicesPlatformSpec `json:"services,omitempty"`
+	// Persistence defines the platform persistence configuration. When this field is set,
+	// the configuration is used as the persistence for platform services and sonataflow instances
+	// that don't provide one of their own.
+	// +optional
+	Persistence *PersistencePlatformSpec `json:"persistence,omitempty"`
+}
+
+// PersistencePlatformSpec configures the DataBase support for both platform services and workflows. For services, it allows
+// configuring a generic database connectivity if the service does not come with its own configured. In case of workflows,
+// the operator will add the necessary JDBC properties to in the workflow's application.properties so that it can communicate
+// with the persistence service based on the spec provided here.
+// +optional
+type PersistencePlatformSpec struct {
```

Comment:
I think `PersistenceOptionsSpec` makes more sense because this is used in both the platform and workflow objects.

Reply:
Will rename